jcl/src/java.base/share/classes/java/lang/Throwable.java

```diff
@@ -319,12 +313,24 @@ public void printStackTrace (PrintStream err) {
  *				The writer to write the walkback on.
  */
 public void printStackTrace(PrintWriter err) {
+	printStackTraceHelper(err);
+}
+
+/**
+ * Outputs representation of the receiver's
+ * walkback on the Appendable specified by the argument.
+ *
+ * @param	appendable Appendable 
+ *		The Appendable object to the walkback will be written.
+ */
+private void printStackTraceHelper(Appendable appendable) {
 	StackTraceElement[] stack;
-	stack = printStackTrace(err, null, 0, false);
+	HashSet<Throwable> exceptionChainSet = new HashSet<Throwable>();
```

Comment:
This could be printing a stack for an OOM. This memory allocation needs to catch OOM and proceed with null if there isn't enough memory to allocation the HashSet. Similarly when adding to the HashSet.